openarm/damiao/__init__.py

```diff
@@ -35,532 +66,933 @@ class MotorType(StrEnum):
+        encode_control_vel(self.bus, self.motor_id, params)
 
-    Args:
-        slave_id (int): Slave ID for the target motor.
-        address (RegisterAddress): The register address to read.
+        # Return coroutine from asynchronous decode function
+        return decode_motor_state(self.bus, self.motor_id, self.motor_limits)
 
-    Returns:
-        can.Message: A CAN message that, when sent, requests the register value.
+    def control_pos_force(
+        self, params: PosForceControlParams
+    ) -> Coroutine[Any, Any, MotorState]:
+        """Control motor in position/force mode. Returns coroutine to be awaited.
 
-    """
-    return can.Message(
-        arbitration_id=0x7FF,
-        data=[
-            slave_id & 0xFF,
-            (slave_id >> 8) & 0xFF,
-            _READ_REGISTER_CODE,
-            address,
-            0x00,
-            0x00,
-            0x00,
-            0x00,
-        ],
-        is_extended_id=False,
-    )
-
-
-def write_register_command(
-    slave_id: int, address: RegisterAddress, value: float, *, as_float: bool = False
-) -> can.Message:
-    """Create a CAN message to write a value to a specific register of a Damiao motor.
-
-    Args:
-        slave_id (int): Slave ID of the target motor.
-        address (RegisterAddress): The register address to write.
-        value (float): The value to write to the register.
-        as_float (bool, optional): If True, the value is written as a 32-bit float.
-
-    Returns:
-        can.Message: A CAN message that, when sent, writes the value to the register.
+        Args:
+            params: Position and force control parameters dataclass
 
-    """
-    return can.Message(
-        arbitration_id=0x7FF,
-        data=[
-            slave_id & 0xFF,
-            (slave_id >> 8) & 0xFF,
-            _WRITE_REGISTER_CODE,
-            address,
-            *(
-                struct.pack("<f", float(value))
-                if as_float
-                else struct.pack("<I", int(value))
-            ),
-        ],
-        is_extended_id=False,
-    )
-
-
-def set_control_mode_command(slave_id: int, mode: ControlMode) -> can.Message:
-    """Create a CAN message to set the control mode of a Damiao motor.
-
-    Args:
-        slave_id (int): Slave ID of the target motor.
-        mode (ControlMode): The control mode to set.
-
-    Returns:
-        can.Message: A CAN message that, when sent, sets the motor's control mode.
+        Returns:
+            Coroutine that yields MotorState when awaited
 
-    """
-    return write_register_command(slave_id, RegisterAddress.CTRL_MODE, mode)
+        Reference: README.md Motor class method pattern lines 334-340
+
+        """
+        # Encode position/force control and send request
+        encode_control_pos_force(self.bus, self.motor_id, params)
+
+        # Return coroutine from asynchronous decode function
+        return decode_motor_state(self.bus, self.motor_id, self.motor_limits)
```

Comment:
encode and decode use a different ID:

```suggestion
        # Encode position/force control and send request
        encode_control_pos_force(self.bus, self.slave_id, params)

        # Return coroutine from asynchronous decode function
        return decode_motor_state(self.bus, self.master_id self.motor_limits)
```

Reply:
done